### ISSUE TITLE
Impossible to attach a file with the share menu

### ISSUE THREAD

**Issue by Massedil** [k9mail/k-9#4399]
### Steps to reproduce

- In a file explorer, share a file.
- K9 is proposed in the list of app for sharing.

### Actual behavior
- When K9 is chosen an new composition screen is launched but without the attached file

### Expected behavior
- The file is attached in the new composition screen

Note that selecting the same file from the composition screen works but not from the file explorer (Amaze here).
It was working before the 7 update.

### Environment
K-9 Mail 702 - F-droid
Samsung Galaxy S5 
Android 9 - LineageOS 16

**Comment by cketti** [comment]
Your file manager is using file:// URIs when sharing files. This hasn't been state of the art for years and due to changes in Android support for this among receiving apps will most likely decrease considerably in the future. 
Get a new file manager or tell them to update their app to use content:// URIs when sharing files.

We won't change K-9 Mail to request the storage permission just to be able to support broken apps. However, we could display a dialog informing the user that the sending app is at fault.

**Comment by Massedil** [comment]
Thanks for the explanation !

**Comment by chrisgch** [comment]
Author of Total Commander here:
Our app allows to share files either as content: URLs or file: URLs. The latter is still needed because many older apps simply don't accept content: URLs. I recommend that you accept file: URLs and content: URLs, it's very easy to support both. You just need to call this:

	public static void disableStrictMode() {
		StrictMode.VmPolicy.Builder builder = new StrictMode.VmPolicy.Builder();
		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P)
			builder.detectNonSdkApiUsage();
		StrictMode.setVmPolicy(builder.penaltyLog().build());
	}

**Comment by cketti** [comment]
@chrisgch: I believe accommodating apps that hang on to the old way of doing things is holding the ecosystem back. I'm fine with leaving apps behind to speed things up.

I don't think the snippet is enough. Apps also need to request the storage permission in order to be able to access shared files.

I'm guessing quite a few users will end up finding this issue. Please feel free to add a comment on how Total Commander users can enable sharing via content: URI.

**Comment by chrisgch** [comment]
I disagree that Total Commander is hanging behind. I offer both options to the user so he can send files to both older and newer apps.

You are right that you need request storage permission to open file: urls. But this is very easy, I can post code if you want. It would be much more user friendly to do this.

**Comment by cketti** [comment]
> I disagree that Total Commander is hanging behind. I offer both options to the user so he can send files to both older and newer apps.

Apologies, I didn't mean it like that. Since Total Commander supports both methods it's not one of the apps that "hang[s] on to the old way of doing things". But I assume you're still supporting file: URIs because of apps that do.
My philosophy is "let's break 'compatibility' with those apps and move on". Yours appears to be that of "let's support as many apps as possible". That's fine. But I believe this is taking away a lot of the pressure on developers to update their apps. And in this case, if the user has to know which option to pick, it doesn't appear to offer a great user experience either.
Granted, failing silently is not much better. "Fail and blame" might be frustrating to users. But ultimately that will either force the developer of the other app to update or it will be abandoned by the user. In both cases it's a win for the ecosystem.
